refactor in favor of smaller classes / functions, more individual uni…

jjelschen opened the pull request:
…t tests

* introduced an enum for key actions
* introduced class `WorkspaceNavigationHelper`, providing methods for traversing the workspace tree.
    * this still is not ideal, since the class requires references to the current workspace, as well as the `UIState` instance. It is nice that `Workspace` is kept clean of `UIState` information, but the other way around there is clearly a dependency. Right now, the two classes are treated as more or less fully independent, being coupled together only by `NavigationComponent`.
* moved test data (exemplary workspace tree) into a separate file, `workspace.spec.data.ts`.
* ~set TypeScript version to 2.4.2, the same that is currently being used in [https://github.com/test-editor/test-editor-web](test-editor-web) (in fact, I just noticed there are two entries in its [yarn.lock](https://github.com/test-editor/test-editor-web/blob/a79596151853526d7140df211775903e2a74ca1a/yarn.lock#L5629) for the TypeScript dependency…!)~ TypeScript version is back to 2.3.0, because Angular 4.x does not seem to like TS > 2.3, and updating to Angular 5.x is a more complex matter. Since string enums are not supported in TS 2.3.0, `KeyActions` is now a class with `static readonly` fields, instead.
* broke up long key event dispatching method.

Commit:
refactor in favor of smaller classes / functions, more individual unit tests

gunther-bachmann commented on src/lib/src/common/util/workspace.navigation.helper.spec.ts:
```diff
@@ -0,0 +1,187 @@
+    });
+  });
+
+  describe('previous()', () => {
```
isn't 'previousVisible' now?

gunther-bachmann commented on src/lib/src/common/util/workspace.navigation.helper.spec.ts:
```diff
@@ -0,0 +1,187 @@
+  });
+
+
+  describe('next()', () => {
```
isn't it 'nextVisible' now?

gunther-bachmann commented on .travis.yml:
```diff
@@ -1,5 +1,7 @@
 dist: trusty
 sudo: false
+addons:
```
should probably use fix/chrome_startup coding in travis ... so this can be removed

jjelschen commented:
addressing remarks: 
* rolled back the change to `.travis.yml`, and rebased to master to incorporate Chrome startup fix
* renamed the two `describe` blocks to match the names of the functions they are testing.
    * https://github.com/test-editor/web-workspace-navigator/blob/b480b36e8ceed60481053b8e8bc84cde2279f434/src/lib/src/common/util/workspace.navigation.helper.spec.ts#L21
    * https://github.com/test-editor/web-workspace-navigator/blob/b480b36e8ceed60481053b8e8bc84cde2279f434/src/lib/src/common/util/workspace.navigation.helper.spec.ts#L103
